src/front/wgsl/mod.rs

```diff
@@ -838,14 +838,14 @@ impl Parser {
                 if let Some(expr) = self.parse_function_call_inner(lexer, word, ctx.reborrow())? {
                     expr
                 } else {
-                    let inner = self.parse_type_decl_impl(
+                    let ty = self.parse_type_decl_impl(
```

Comment:
We should just call `parse_type_decl_name` here, which gives us `Handle`. Then we can do `type_arena[ty].inner` for that match below